### PR TITLE
Implement TraceContext header

### LINKED ISSUE #43: Implement TraceContext header formatter
https://github.com/TraceContext/tracecontext-spec/blob/master/trace_context/HTTP_HEADER_FORMAT.md

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #43 

Had to switch the internal storage of the spanId and parentSpanId to be a hex string because some versions of PHP still run 32-bit (so an integer won't fit). This also affects the interface for the extension (which is still pre-alpha so no big deal).

### CODE REVIEW

**Comment by chingor13** [comment]
Doh, looks like we can't be storing the span id as an int as we're testing against 32-bit versions of PHP as well and the span id is a 64-bit integer (8 bytes).

File: src/Trace/Propagator/TraceContextFormatter.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+use OpenCensus\Trace\TraceContext;
+
+/**
+ * This format using a human readable string encoding to propagate TraceContext.
+ * See https://github.com/TraceContext/tracecontext-spec/blob/master/trace_context/HTTP_HEADER_FORMAT.md
+ * for the definition.
+ */
+class TraceContextFormatter implements FormatterInterface
+{
+    const CONTEXT_HEADER_FORMAT = '/([0-9a-f]{2})-(.*)/';
```
Comment (tmatsuo) [inline src/Trace/Propagator/TraceContextFormatter.php]: Is it possible that the hex values uses uppercase letters?

File: src/Trace/Propagator/TraceContextFormatter.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        return new TraceContext();
+    }
+
+    /**
+     * Convert a TraceContext to header string. Uses version 0.
+     *
+     * @param TraceContext $context
+     * @return string
+     */
+    public function serialize(TraceContext $context)
```
Comment (tmatsuo) [inline src/Trace/Propagator/TraceContextFormatter.php]: How do you add another serializer when the version 1 is released?

Comment (chingor13) [reply src/Trace/Propagator/TraceContextFormatter.php]: Most likely, we will add an option in the constructor for which version to output.

File: src/Trace/Propagator/TraceContextFormatter.php (hunk 1 above)
Comment (chingor13) [reply src/Trace/Propagator/TraceContextFormatter.php]: It's not excluded in the spec, so I'll make sure it works for both.

**Pull request merged**